### PR TITLE
feat: MPI operator installation code for distributed training use case

### PR DESCRIPTION
**Pull request by sanjeevrg89**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

I have added MPI Operator installation. The goal is to use MPI Operator to run allreduce-style distributed training on Amazon EKS.

### Motivation
I am working on a blog that leverages MPI operator for distributed training.

### More

- [ Yes] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ No] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ No] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [Yes ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes
Screenshot 2023-11-05 at 7 34 43 PM

I have tested the installation on Amazon EKS 1.27 cluster, MPI operator installation works great.

### CODE REVIEW

**Comment by sanjeevrg89** [comment]
Made MPI operator as an optional install for users

File: ai-ml/trainium-inferentia/variables.tf [hunk 1]
```diff
@@ -37,3 +37,14 @@ variable "enable_amazon_prometheus" {
   type        = bool
   default     = true
 }
+
+variable "mpi_operator_version" {
+  description = "The version of the MPI Operator to install"
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/variables.tf:42]: Add type as string

**Review by vara-bonthu — COMMENTED** [review]
Thanks for adding mpi operator. One minor comment

File: ai-ml/trainium-inferentia/variables.tf (hunk 1 above)
Comment (sanjeevrg89) [reply ai-ml/trainium-inferentia/variables.tf:42]: @vara-bonthu Thank you for the feedback. I added type as string in the variable

**Review by vara-bonthu — APPROVED** [review]

**Pull request merged**